components/datadog/agent/host.go

```diff
@@ -37,7 +37,7 @@ func (h *HostAgent) Export(ctx *pulumi.Context, out *HostAgentOutput) error {
 }
 
 // NewHostAgent creates a new instance of a on-host Agent
-func NewHostAgent(e *config.CommonEnvironment, host *remoteComp.Host, options ...agentparams.Option) (*HostAgent, error) {
+func NewHostAgent(e *config.CommonEnvironment, host *remoteComp.Host, dep pulumi.Resource, options ...agentparams.Option) (*HostAgent, error) {
```

Comment:
You should not add parameter here but add a `WithPulumiResourceOptions` in `agentparams` (similar to https://github.com/DataDog/test-infra-definitions/blob/main/components/datadog/dockeragentparams/params.go#L81-L86 - but with proper name).